pom.xml

```diff
@@ -475,6 +479,48 @@
             <argLine>-Xmx1024m -XX:MaxPermSize=128m -Dfile.encoding=UTF-8</argLine>
           </configuration>
         </plugin>
+        <plugin>
+          <groupId>org.jacoco</groupId>
+          <artifactId>jacoco-maven-plugin</artifactId>
+          <version>0.7.5.201505241946</version>
+          <executions>
+            <execution>
+              <id>default-prepare-agent</id>
+              <goals>
+                <goal>prepare-agent</goal>
+              </goals>
+            </execution>
+            <execution>
+              <id>default-report</id>
+              <phase>prepare-package</phase>
+              <goals>
+                <goal>report</goal>
+              </goals>
+            </execution>
+            <execution>
+              <id>default-check</id>
+              <goals>
+                <goal>check</goal>
+              </goals>
+              <configuration>
+                <!-- Do not fail the build if rules are not met -->
+                <haltOnFailure>false</haltOnFailure>
+                <rules>
+                  <rule>
+                    <element>BUNDLE</element>
+                    <limits>
+                      <limit>
+                        <counter>LINE</counter>
+                        <value>COVEREDRATIO</value>
+                        <minimum>0.9</minimum>
```

Comment:
Yes, it is strict. We could certainly have a long discussion about the rules... I came across a statement that I liked:

> There is no reason to have less than 95% of lines covered by tests when doing TDD.

Or something like that. I think it's true. Rhetorical question: Do we really have code that is never executed during testing? Are we OK about that?

On the other hand, who is doing TDD here? Additionally, there is some code where the added value of tests is questionable (e.g. GWT client code, maybe some other...).

So, this rule is basically a demo that will print a warning with the actual line coverage (if it's below 90%) so anyone who's running a build can either investigate the report and maybe make some improvement or just ignore it. I think there is no "one size fits all" so if we want the `check` goal to be taken seriously, or even have it fail the build, rules will have to be tailored in different ways for different projects.